docs/chapter-5.mdx

```diff
@@ -1,90 +1,205 @@
+
+## What does an Action look like?
+
+Actions are simple. An action is usually an object with just two properties: a `type`, a string that indicates what _exactly_ that action represents, and an optional `payload` that is kind of like a function argument. In our example, an action that adds a category might look like this:
+
+```ts
+const addCategoryAction = {
+  type: "[Categories List] Add Category",
+  payload: { name: "Food" },
+};
+```
+
+This is an action that adds a category named `Food` to the list of all categories. Of course, we still haven't written the logic that actually uses this action to put the data in the store, but for now we are focusing on the `Actions`.
+
+
+### Creating an action
+
+NgRx provides some utility functions to help us create actions instead of creating our objects by hand as we did in the previous example.
+
+To start learning about those, let's create a folder names `state` in the `src/app` directory, and a file called `actions.ts` inside it.
+
+#### Using `createAction`
+
+The first way of doing so is by using `createAction` which will create a single action given the name and the parameters provided:
+
+```ts title="state/actions.ts"
+import { createAction, props } from "@ngrx/store";
+import { Category } from "./state";
+
+export const addCategory = createAction(
+  "[Category List] Add Category",
+  props<{ category: Category }>()
+);
+```
+
+Let's break down this example. First of all, the name `createAction` is a bit deceptive; it does not create an action; in fact, it creates a function which, when called, wil produce an action object. The first argument is the `type` of the action that will be produced. When called, the `addCategory` function will **always** create an action with type "[Category List] Create Category". The second argument uses the bizarre function `props`, which is a generic function that allows us to define the type of the `payload` which the created action will have. Essentially, it explains that in order to create the action using the `addCategory` function, we should call it and provide an object that has a property `category` which is defined in the Interface Category defined in the AppState. Let's do this and `console.log` the result.
+
+```ts title="app.component.ts"
+import { Component, OnInit } from "@angular/core";
+import { addCategory } from "./state/actions";
+
+@Component({/* ... */})
+export class AppComponent implements OnInit {
+  ngOnInit() {
+    console.log(addCategory({category:{ name: "Food" }}));
+  }
+}
+```
+
+In the console, we will see the following:
+
+```js
+{category: {name: 'Food', type: '[Category List] Add Category'}}

```

Comment:
I think this was wrong to begin with - shouldn't the `type` be in the first level of the object together with `category`?

Reply:
Indeed, I blindly copy/pasted it. I'm fixing this right away!